src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1100,10 +1100,17 @@ var ReactCompositeComponent = {
    * @final
    * @private
    */
-  attachRef: function(ref, component) {
+  attachRef: function(ref, component, mockConfig) {
     var inst = this.getPublicInstance();
     invariant(inst != null, 'Stateless function components cannot have refs.');
-    var publicComponentInstance = component.getPublicInstance();
+    var publicComponentInstance;
+    if (mockConfig && mockConfig.getMockRef) {
```

Comment:
>one issue is that component.getPublicInstance() will be calling ReactCompositeComponent.getPublicInstance if the component is a composite.

I think it’s reasonable to scope `getMockRef` effect to host instances only. You don’t want to mock composite refs, I think.